cinema/views.py

```diff
@@ -81,12 +107,25 @@
         MovieSession.objects.all()
         .select_related("movie", "cinema_hall")
         .annotate(
-            tickets_available=F("cinema_hall__rows")
-            * F("cinema_hall__seats_in_row")
-            - Count("tickets")
+            tickets_available=F("cinema_hall__rows") * F(
+                "cinema_hall__seats_in_row"
+            ) - Count("tickets")
         )
     )
     serializer_class = MovieSessionSerializer
+    permission_classes = (IsAdminOrIfAuthenticatedReadOnly,)
+
+    def get_permissions(self):
+        if self.action in (
+                "list",
+                "create",
+                "retrieve",
+                "update",
+                "partial_update",
+                "destroy"
+        ):
+            return (IsAdminOrIfAuthenticatedReadOnly(),)
+        return (IsAuthenticated(),)
```

Comment:
The `get_permissions` method in `MovieSessionViewSet` allows actions like 'create', 'update', and 'destroy' for authenticated users. Review and adjust these permissions to ensure they align with the application's access control policy, as these actions might be intended only for admins.